xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -612,7 +612,7 @@ float CGUIFontTTF::GetTextWidthInternal(const vecText& text, std::vector<Glyph>&
       // If last character in line, we want to add render width
       // and not advance distance - this makes sure that italic text isn't
       // choped on the end (as render width is larger than advance then).
-      if (it == glyphs.end())
+      if (it == glyphs.end() - 1)
```

Comment:
hmmmm, in case the vector glyphs is empty this for-loop would never be entered! still i have a feeling that `end() - 1` in general is quite ugly and unsafe. can somebody tell me the bulletproof solution please?

Reply:
std::prev(glyphs.end())?

Reply:
`glyphs.back()`

Reply:
or `glyphs.rbegin()`

Reply:
turns out that `rbegin()` doesnt compile: `no match for ‘operator==’ (operand types are ‘__normal_iterator<>’ and ‘::reverse_iterator’`.

`back()` doesnt return an iterator but the element.. :( so we can't `*it == back()`, the first `n` in the example screenshots would calculate wrong in that case.

`std::prev(glyphs.end())` same issue like `end()-1` with an empty vec. ----
well, inside this loop it is impossible that the vector is empty.. the winner is `(std::next(it) == end())`.. makes all no difference. a mind game. i should not overthink it ;)